standard/region.lua

```diff
@@ -0,0 +1,52 @@
+local region = {}
+local Class = require('Module:Class')
+local Flag = require('Module:Flags')
+local String = require('Module:String')
+local Lua = require('Module:Lua')
+local regionData = mw.loadData('Module:Region/Data')
+local countryToRegionData = Lua.loadDataIfExists('Module:Region/CountryData', {})
+
+function region.run(args)
+	args = args or {}
+	local regionEntry = args.region
+	local onlyRegion = args.onlyRegion == 'true'
+	local onlyDisplay = args.onlyDisplay == 'true'
+
+	if String.isEmpty(regionEntry) then
+		local country = Flag._CountryName(args.country)
+		regionEntry = countryToRegionData[string.lower(country)]
+		if String.isEmpty(regionEntry) then
+			return ''
+		end
+	end
+
+	regionEntry = string.lower(regionEntry)
+	regionEntry = regionData.aliases[regionEntry] or regionEntry
+	local regionReturn = regionData[regionEntry] or {}
+	if onlyRegion then
+		return regionReturn.region or ''
+	elseif regionReturn ~= {} then
```

Comment:
```suggestion
	elseif not Table.isEmpty(regionReturn) then
```

Reply:
would always return false as the table function uses `next` to determine if the table is empty, but since we have a table with named keys (not numbered keys) here this wouldn't work

Reply:
next works with both named and numbered keys
```
next({a = 3}, nil)
-- returns a, 3
next({3}, nil)
-- returns 1, 3
```

Reply:
and yet i still get it telling me the table is empty even though it is not

![Screenshot 2021-09-28 14 41 50](https://user-images.githubusercontent.com/75081997/135089005-1fbc2bcf-51a5-4070-8e49-374d1a803976.png)

maybe it is the metatable it doesn't like

Reply:
Yea I've experienced this too with the `isEmpty` function, we should fix it.

Reply:
using `setmetatable(tbl, nil)` doesn't work due to "cannot change a protected metatable" (the metattable from `mw.loadData` apparently is protected)

tested a few alternatives --> https://liquipedia.net/starcraft2/Module:Hjpalpha/sandbox15
of those the only one that catches the metatable stuff correctly would be
```lua
function p.isEmpty4(tbl)
	if tbl == nil then
		return true
	end
	for _, _ in pairs(tbl) do
		return false
	end
	return true
end
```
keep in mind that this will see
```lua
{
	{}
}
```
as not empty (like with the old implementation)

if you like that one i would add it tomorrow
else we would have to search for a better solution

Reply:
changed the code in the region module a bit (adding a fall back)
this results in the `table.isEmpty` not being needed anymore, but we should still adjust the table module i think